torchdata/dataloader2/reading_service.py

```diff
@@ -300,7 +301,6 @@ def initialize_iteration(self) -> None:
             dist_info = _DistInfo(shared_seed_int, self._world_size, self._rank)
             call_on_epoch_reset = partial(process_reset_fn, dist_info=dist_info, custom_reset_fn=self.worker_reset_fn)
             end_datapipe.reset_epoch(call_on_epoch_reset)
-            end_datapipe.reset()
```

Comment:
Do we need to call this when `self.main_prefetch_cnt <= 0`?

Reply:
Not needed because `reset` is always called for each iteration of datapipe.